uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -271,14 +271,16 @@ define([
 
       /**
        * Checks if given value is added for leave request list of document value ie., field required_documents
-       *  otherwise add it to list of required documents
+       *  otherwise add it to list of required documents (list is actually string of comma separated values for now)
        *
        * @param {String} value
        */
       vm.updateRequiredDocuments = function (value) {
-        var index = vm.leaveRequest.required_documents.indexOf(value);
+        var docsArray = vm.leaveRequest.required_documents ? vm.leaveRequest.required_documents.split(','): [];
+        var index = docsArray.indexOf(value);
 
-        index === -1 ? vm.leaveRequest.required_documents.push(value) : vm.leaveRequest.required_documents.splice(index, 1);
+        index === -1 ? docsArray.push(value) : docsArray.splice(index, 1);
+        vm.leaveRequest.required_documents = docsArray.join(',');
```

Comment:
I think that this whole logic for removing/adding a document to the `required_documents` list should be encapsulated in a `SicknessRequestInstance` method. Something like
```js
// This adds/removes the given document from the `required_documents` list
sicknessRequest.toggleDocument(document);
```
so that the controller shouldn't care about how the instance stores the list of documents